app/src/lib/stores/git-store.ts

```diff
@@ -320,6 +324,22 @@ export class GitStore extends BaseStore {
     return allBranchesWithUpstream
   }
 
+  private async checkPullWithRebase() {
+    const result = await getGlobalConfigValue('pull.rebase')
```

Comment:
I worry about how often this will fire to keep it up to date, and whether there's a better way to get the global config and cache in app state. We don't use this value when it's time to perform the pull, but it's needs to be checked more frequently than our other uses in Preferences.